R/cql-translator.R

```diff
@@ -11,21 +11,33 @@
 # See the License for the specific language governing permissions and limitations under the License.
 
 # Function to translate R code to CQL
-cql_translate <- function(...) {
+cql_translate <- function(dots) {
+  ## dots should be a list of quosures coming from filter
+  ## run partial_eval on them to evaluate named objects
+  ## e.g., if x is defined in the global env and passed as on object to
+  ## filter, need to evaluate x.
+  dots <- lapply(dots, function(x) {
+    rlang::new_quosure(
+      dbplyr::partial_eval(rlang::get_expr(x), env = rlang::get_env(x)),
+                rlang::get_env(x))
+  })
   # when there are spatial predicates (e.g., DWITHIN, INTERSECTS, TOUCHES)
-  # evaluate those parts of the expression so they are expanded
-  dots <- expand_spatial_predicates(...)
+  # evaluate those parts of the expression so they are expanded.
+  dots <- expand_spatial_predicates(dots)
```

Comment:
I'm actually fairly certain this is now unnecessary, as I think the partial eval above does this for us

Reply:
Confirmed and removed in 5251cb5